<?xml version="1.0"?>
<launch>
  <!-- settings -->
  <param name="sim_mode" value="false" />
  <param name="debug_mode" value="false" />
  <param name="quad_frame" value="NWU" />
  <param name="estimate_frame" value="INERTIAL" />

  <!-- mavros -->
  <node pkg="atl_ros" name="atl_fake_mavros" type="fake_mavros_node" output="screen" required="true" />

  <!-- controller node -->
  <param name="/control/config_dir" value="$(find atl_configs)/configs/quadrotor_anastasia" />
  <node pkg="atl_ros" name="atl_control" type="control_node" output="screen" required="true" />

  <!-- apriltag node -->
  <param name="/apriltag/config" value="$(find atl_configs)/configs/apriltag/config.yaml" />
  <node pkg="atl_ros" name="atl_apriltag" type="apriltag_node" output="screen" required="true" />

  <!-- estimate node -->
  <param name="tracker_mode" value="1" />
  <param name="/estimator/kf/config" value="$(find atl_configs)/configs/estimator/kf_tracker.yaml" />
  <param name="/estimator/ekf/config" value="$(find atl_configs)/configs/estimator/ekf_tracker.yaml" />
  <node pkg="atl_ros" name="atl_estimator" type="estimator_node" output="screen" required="true" />

  <!-- pg camera node -->
  <param name="camera_config_dir" value="$(find atl_configs)/configs/camera/pointgrey_chameleon" />
  <node pkg="atl_ros" name="atl_camera" type="pg_camera_node" output="screen" required="true" />

  <!-- gimbal node -->
  <param name="/gimbal/config" value="$(find atl_configs)/configs/gimbal/config.yaml" />
  <node pkg="atl_ros" name="atl_gimbal" type="gimbal_node" output="screen" required="true" />

  <!-- imu node -->
  <node pkg="atl_ros" name="atl_imu" type="imu_node" output="screen" launch-prefix="sudo -E" required="true" />
</launch>
